_posts/2025-05-07-release-notes-3.7.0.md

```diff
@@ -0,0 +1,522 @@
+Previously expression compiler was developed independently from the Scala compiler inside [scalacenter/scala-debug-adapter](https://github.com/scalacenter/scala-debug-adapter) repository and was cross-published for every Scala release.
+Starting with Scala 3.7 the expression compiler has been migrated to the main [scala/scala3](https://github.com/scala/scala3) repository and become an integral part of Scala toolchain. This change allows users to use the expression compiler with nightly versions of the compiler as well, easing the maintenance and release process for the compiler team.
+
+### Presentation Compiler: Show inferred type on holes in hover ([#21423](https://github.com/scala/scala3/pull/21423))
+
+The presentation compiler is a special mode of the Scala compiler that runs interactively and is used by IDEs and language servers such as Metals. It provides immediate feedback about code correctness, type checking, symbol resolution, autocompletion, error highlighting, and other editing support functionalities.
+Some of the latest improvements to the presentation compiler focus on the ability to infer more information about expected types of expressions provided by the users. As a result, presentation compiler can now show the users the expected type of expression when hovering over so called `type holes`.
+
+```scala
+def someMethod(count: Int, label: String): Unit = ???
+someMethod(???, ???)
+```
+
+Given the code snippet above, hovering on either placeholder will hint about the expected type (`Int` and `String`, respectively).
+
+### Quotes API changes
+
+Scala 3.7 introduces changes to the experimental subset of Quotes API. One notable addition is the `apply` methods to import selectors ([#22457](https://github.com/scala/scala3/pull/22457)), allowing to dynamically construct import statements within macros. This feature can be especially useful when summoning implicit instances in the generated code, as it allows to add additional instances to the search​.
+
+Another advancement is the experimental `summonIgnoring` method ([#22417](https://github.com/scala/scala3/pull/22417)). It allows developers to summon implicit instances while excluding specific instances of givens from the search. It's particularly useful in complex derivation scenarios where certain implicit should be disregarded to prevent conflicts or unintended resolutions.​ It might be used to provide a custom implementation of macro only in cases where a user-defined variant of implicit is available, while avoiding recursively re-calling the macro method.
+
+```scala
+//> using options -experimental
+
+import scala.quoted.*
+
+trait ExcludedType
+trait Show[T]:
+  def show(): Unit
+
+object Show {
+  implicit transparent inline def auto[T]: Show[T] = ${ autoImpl[T] }
```

Comment:
Is it `implicit transparent inline def` instead of `transparent inline given` on purpose?